sw/host/opentitanlib/src/spiflash/sfdp.rs

```diff
@@ -38,14 +40,20 @@ pub struct SfdpHeader {
 impl TryFrom<&[u8]> for SfdpHeader {
     type Error = Error;
     fn try_from(buf: &[u8]) -> Result<Self, Self::Error> {
+        const SFDP_HEADER_SIGNATURE: u32 = 0x50444653;
+
         let mut reader = std::io::Cursor::new(buf);
-        Ok(SfdpHeader {
+        let header = SfdpHeader {
             signature: reader.read_u32::<LittleEndian>()?,
             minor: reader.read_u8()?,
             major: reader.read_u8()?,
             nph: reader.read_u8()?,
             reserved: reader.read_u8()?,
-        })
+        };
+        if header.signature != SFDP_HEADER_SIGNATURE {
+            return Result::Err(Error::WrongHeaderSignature(header.signature));
+        }
+        Ok(header)
```

Comment:
```suggestion
    match header.signature {
        SFDP_HEADER_SIGNATURE => Ok(header),
        v => Err(WrongHeaderSignature(v)),
    }
```

Reply:
Ah! C++ has rotted my brain :P

Reply:
Haha, suffering from the same ailment :)